lib/improver/tests/ensemble_calibration/ensemble_calibration/helper_functions.py

```diff
@@ -38,9 +38,85 @@
 import iris
 from iris.coords import AuxCoord, DimCoord
 from iris.cube import Cube, CubeList
+from iris.tests import IrisTest
 import numpy as np
 
+from improver.tests.set_up_test_cubes import set_up_variable_cube
 from improver.utilities.cube_manipulation import concatenate_cubes
+from improver.utilities.warnings_handler import ManageWarnings
+
+
+IGNORED_MESSAGES = ["Collapsing a non-contiguous coordinate.",
+                    "Not importing directory .*sphinxcontrib'",
+                    "The pandas.core.datetools module is deprecated",
+                    "numpy.dtype size changed",
+                    "The statsmodels can not be imported",
+                    "invalid escape sequence",
+                    "can't resolve package from",
+                    "Minimisation did not result in"
+                    " convergence",
+                    "\nThe final iteration resulted in a percentage "
+                    "change that is greater than the"
+                    " accepted threshold ",
+                    "divide by zero encountered in true_divide",
+                    "invalid value encountered in"]
+WARNING_TYPES = [UserWarning, ImportWarning, FutureWarning, RuntimeWarning,
+                 ImportWarning, DeprecationWarning, ImportWarning, UserWarning,
+                 UserWarning, RuntimeWarning, RuntimeWarning]
+
+
+class SetupCubes(IrisTest):
+
+    """Set up cubes for testing."""
+
+    @ManageWarnings(
+        ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
+    def setUp(self):
+        """Set up temperature and wind speed cubes for testing."""
```

Comment:
Do we necessarily need both wind and temperature cubes - especially when it seems like the numbers in the arrays are identical? We could halve the number of lines of code if we only need one.

Reply:
Yes, this is a fair point. The main reason to have both the wind and temperature cubes is because they use a slightly different CRPS minimisation function underneath i.e. Gaussian and truncated Gaussian. The aim was to test both of these minimisation functions from the top level, so including estimating and applying the coefficients. I guess the concern I could see might be that people might have less confidence in one of the minimisation functions, if only the other minimisation function is fully tested. I haven't removed any of these tests now, but we could have a chat about it.

Reply:
In that case, we're really testing technical functionality (e.g: the different CRPS minimization function), rather than a physical thing - I think references to physical observables are just adding unnecessary code in my opinion.

I'd generally just go for a `diagnostic='foo'` type approach, have only a single cube, and then just change which of the minimization functions we're using, but with the same cube. We can discuss face-to-face if this isn't clear though. Maybe I'm just misunderstanding the problem. :)